docs/en/extn-developer-FAQ.md

```diff
@@ -0,0 +1,93 @@
+# Extension Developer FAQ
+
+**# # # Work in Progress # # #** 
+
+## How does the non-OS SDK structure execution
+
+Apologies for this long description, but none of this is documented by
+Espressif. So this summarises our understanding of how **non-OS SDK** execution
+model works based on the Espressif-supplied examples and SDK documentation, plus
+various posts on the Espressif BBS and other forums, and an examination of the
+BootROM code.
+
+The ESP8266 boot ROM contains a set of primitive tasking and dispatch functions
+which are also used by the SDK. In this model, execution units are either:
+
+-   **INTERRUPT SERVICE ROUTINES (ISRs)** which are declared and controlled
+    through the `ets_isr_attach()` and other `ets_isr_*` and `ets_intr_*`
+    functions. ISRs can be defined on a range of priorities, where a higher
+    priority ISR is able to interrupt a lower priority one. ISRs are time
+    critical and should complete in no more than 50 µSec.
+  
+    ISR code and data constants should be run out of RAM or ROM, as the
+    execution time for Flash memory (that is located in the `irom0` load section)
+    is indeterminate: whilst cache-hits can run at full memory bandwidth, any
+    cache-misses require the code to be read from Flash; and even though
+    H/W-based, this is at roughly 26x slower than memory bandwidth (for DIO
+    flash); this will cause ISR execution to fall outside the required time
```

Comment:
As I understand, the real problem is not that it is slower, but that it (potentially) interferes with loading code out of flash that happens at the base (non-interrupt) level. This makes it absolutely wrong, rather than -- you will probably violate some timing constraint..